File: day1/day1.ts
const part1 = (input: string[]) => {
  const result = input.map(v => +v).reduce((t, v, i, a) => v > a[i - 1] ? t + 1 : t, 0);
  console.log('    Part 1:', result.toString(10));
}

const part2 = (input: string[]) => {
  const result = input
    .map(v => +v)
    .map((_, i, a) => a[i] + a[i + 1] + a[i + 2])
    .slice(0, -2)
    .reduce((t, v, i, a) => v > a[i - 1] ? t + 1 : t, 0);
  console.log('    Part 2:', result.toString(10));
}

const input = Deno.readTextFileSync("input.txt").split('\n');

console.log("Deno:")

part1(input);
part2(input);

File: day1/day1.java
package day1;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class day1 {
  public static void main(String[] args) {
    System.out.println("Java: ");
    try {

      File myObj = new File("input.txt");

      System.out.println("    Part 1: " + day1.part1(myObj).toString());
      System.out.println("    Part 2: " + day1.part2(myObj).toString());
    } catch (FileNotFoundException e) {
      System.out.println("Input file not found");
    }
  }

  public static Integer part1(File file) throws FileNotFoundException {
    Integer prevValue = 1000000;
    Integer total = 0;

    Scanner myReader = new Scanner(file);
    while (myReader.hasNextLine()) {
      String data = myReader.nextLine();
      Integer inputValue = Integer.parseInt(data);

      if (inputValue > prevValue) {
        total += 1;
      }

      prevValue = inputValue;
    }
    myReader.close();
    return total;
  }

  public static Integer part2(File file) throws FileNotFoundException {
    Integer prevValue1 = 1000000;
    Integer prevValue2 = 1000000;
    Integer prevValue3 = 1000000;
    Integer total = 0;

    Scanner myReader = new Scanner(file);
    while (myReader.hasNextLine()) {
      String data = myReader.nextLine();
      Integer inputValue = Integer.parseInt(data);

      if (inputValue + prevValue1 + prevValue2 > prevValue1 + prevValue2 + prevValue3) {
        total += 1;
      }

      prevValue3 = prevValue2;
      prevValue2 = prevValue1;
      prevValue1 = inputValue;
    }
    myReader.close();
    return total;
  }
}
